.github/workflows/kyber.yml

```diff
@@ -0,0 +1,105 @@
+name: Kyber Tests
+
+on:
+  push:
+    branches: [ '*' ]
+  pull_request:
+    branches: [ '*' ]
+
+jobs:
+  build_liboqs:
+    name: Build liboqs
+    runs-on: ubuntu-latest
+    timeout-minutes: 4
+    steps:
+      - name: Checking cache for liboqs
+        uses: actions/cache/restore@v4
+        id: cache-liboqs
+        with:
+          path: opt/liboqs/
+          key: wolfssh-liboqs-0.10.0-${{ runner.os }}
+          lookup-only: true
+
+      - name: Checkout liboqs
+        if: steps.cache-liboqs.outputs.cache-hit != 'true'
+        uses: actions/checkout@v4
+        with:
+          repository: open-quantum-safe/liboqs
+          ref: 0.10.0
+          path: liboqs
+
+      - name: Build and install liboqs
+        if: steps.cache-liboqs.outputs.cache-hit != 'true'
+        working-directory: liboqs
+        run: |
+          mkdir build
+          cd build
+          cmake -DCMAKE_INSTALL_PREFIX=${{ github.workspace }}/opt/liboqs -DOQS_MINIMAL_BUILD=KEM_kyber_512 -DOQS_USE_OPENSSL=0 ..
+          make
+          make install
+
+      - name: Stash liboqs in cache
+        if: steps.cache-liboqs.outputs.cache-hit != 'true'
+        uses: actions/cache/save@v4
+        with:
+          path: opt/liboqs/
+          key: wolfssh-liboqs-0.10.0-${{ runner.os }}
+
+  build_wolfssl:
+    name: Build wolfssl
+    runs-on: ubuntu-latest
+    timeout-minutes: 4
+    steps:
+      - name: Checking cache for wolfssl
+        uses: actions/cache/restore@v4
+        id: cache-wolfssl
+        with:
+          path: opt/wolfssl/
+          key: wolfssh-wolfssl-v5.7.0-stable-${{ runner.os }}
+          lookup-only: true
+
+      - name: Checkout, build, and install wolfssl
+        if: steps.cache-wolfssl.outputs.cache-hit != 'true'
+        uses: wolfSSL/actions-build-autotools-project@v1
+        with:
+          repository: wolfssl/wolfssl
+          ref: v5.7.0-stable
+          path: wolfssl
+          configure: --prefix=${{ github.workspace }}/opt/wolfssl --enable-wolfssh --enable-cryptonly --disable-examples --disable-crypttests
```

Comment:
The prefix won't work because the action sets its own prefix https://github.com/wolfSSL/actions-build-autotools-project/blob/905111b5a8be134fed11e01362427534f36febb7/action.yml#L85. The location is `$GITHUB_WORKSPACE/build-dir`. You can move the dir after this step if you still want it at `$GITHUB_WORKSPACE/opt/wolfssl`.